docs/src/lints/roblox_react_exhaustive_deps.md

```diff
@@ -0,0 +1,88 @@
+# roblox_react_exhaustive_deps
+
+## What it does
+Checks that all dependencies used inside React/Roact hooks like `useEffect`, `useCallback`, and `useMemo` are correctly specified in the dependency array.
+
+## Why this is bad
+Missing dependencies can cause your effects or memoized values to use stale data, leading to bugs. Unnecessary dependencies can cause unnecessary re-renders or re-computations, hurting performance.
+
+When you reference a variable inside a React hook callback, React expects you to declare it as a dependency so it knows when to re-run the effect or recompute the value. Failing to do so can lead to:
+
+1. **Stale closures**: The callback captures the variable's value from when it was created, not its current value
+2. **Missing updates**: Changes to dependencies won't trigger the effect to re-run
+3. **Hard-to-debug issues**: The behavior might work initially but break when component re-renders occur in different orders
+
+## Example
+
+### Missing dependencies
+```lua
+local React = require(Packages.React)
+
+local function Component(props)
+    -- Bad: props.userId is used but not in dependencies
+    React.useEffect(function()
+        fetchUser(props.userId)
+    end, {})
+    
+    -- Good: all dependencies are listed
+    React.useEffect(function()
+        fetchUser(props.userId)
+    end, { props.userId })
+end
+```
+
+### Unnecessary dependencies
+```lua
+-- Bad: count is not used in the effect
+React.useEffect(function()
+    print("Hello")
+end, { count })
+
+-- Good: empty dependencies since nothing is used
+React.useEffect(function()
+    print("Hello")
+end, {})
+```
+
+### Property access dependencies
+```lua
+-- Bad: props.user.name is used but not in dependencies
+React.useEffect(function()
+    setTitle(props.user.name)
+end, {})
+
+-- Good: property access is correctly tracked
+React.useEffect(function()
+    setTitle(props.user.name)
+end, { props.user.name })
+```
+
+## Supported hooks
+This lint checks the following React/Roact hooks:
+- `useEffect` - dependencies are the second parameter
+- `useLayoutEffect` - dependencies are the second parameter  
+- `useCallback` - dependencies are the second parameter
+- `useMemo` - dependencies are the second parameter
+- `useImperativeHandle` - dependencies are the third parameter
+
+## Remarks
```

Comment:
How does this handle `a.b()`? There were some callouts around this in my implementation https://github.com/Kampfkarren/selene/pull/548/files#diff-a6f098e5f68aea9be9769adc9f7b4fc4d3c1923ded1e667ccca6d32ccee0d703R59-R62